examples/HelloDepthStencil.cpp

```diff
@@ -0,0 +1,352 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "Utils.h"
+
+#include <vector>
+#include <glm/glm/glm.hpp>
+#include <glm/glm/gtc/matrix_transform.hpp>
+#include <glm/glm/gtc/type_ptr.hpp>
+
+nxt::Device device;
+
+nxt::Buffer indexBuffer;
+nxt::Buffer vertexBuffer;
+nxt::Buffer planeBuffer;
+nxt::Buffer cameraBuffer;
+nxt::Buffer transformBuffer[2];
+
+nxt::BindGroup cameraBindGroup;
+nxt::BindGroup bindGroup[2];
+nxt::BindGroup cubeTransformBindGroup[2];
+
+nxt::Queue queue;
+nxt::Pipeline pipeline;
+nxt::Pipeline planePipeline;
+nxt::Pipeline reflectionPipeline;
+nxt::RenderPass renderpass;
+nxt::Framebuffer framebuffer;
+
+void initBuffers() {
+    static const uint32_t indexData[6*6] = {
+        0, 1, 2,
+        0, 2, 3,
+
+        4, 5, 6,
+        4, 6, 7,
+
+        8, 9, 10,
+        8, 10, 11,
+
+        12, 13, 14,
+        12, 14, 15,
+
+        16, 17, 18,
+        16, 18, 19,
+
+        20, 21, 22,
+        20, 22, 23
+    };
+    indexBuffer = device.CreateBufferBuilder()
```

Comment:
nit: we could have a helper function in Utils that given a (device, pointer, size, final usage) creates a buffer with that single usage and data.